File: vendedor/api/models/Cliente.java
public class Cliente {
    private int id;
    private String nome;
    private String email;
    private String endereco;
    private String senha;
    private String urlAvatar;
    private String cargo;

    public Cliente(
        int id,
        String nome, String email, String endereco, String senha,
        String urlAvatar, String cargo
    ) {
        if(id <= 0)
            throw new IllegalArgumentException("id não pode ser menor ou igual a 0");
        if(nome == null)
            throw new IllegalArgumentException("nome não pode ser nulo");
        if(email == null)
            throw new IllegalArgumentException("nome não pode ser nulo");
        if(endereco == null)
            throw new IllegalArgumentException("nome não pode ser nulo");
        if(senha == null)
            throw new IllegalArgumentException("nome não pode ser nulo");
        if(cargo == null)
            throw new IllegalArgumentException("nome não pode ser nulo");

        this.id = id;
        this.cargo = cargo;
        this.email = email;
        this.endereco = endereco;
    }
}

File: site/src/services/cliente.ts
import BaseConnect from "./base_connect"

export interface Cliente {
  id: number;
  nome: string;
  email: string;
  endereco: string;
  urlAvatar: string;
  senha: string;
  cargo: string;
}

export interface ClienteToken {

}

export interface ClienteInfo {
  
}

export default class ClienteConnection extends BaseConnect {
  static base_path = '/user';

  static async BuscarTodos() {
    const response = await this.get(`${this.base_path}/list/all`);
    return response;
  }

  static async BuscarPorId(id: number) {
    const response = await this.get(`${this.base_path}/find/:id`, { params: { id } });
    return response;
  }

  static async Criar(data: object) {
    const response = await this.post(`${this.base_path}/`, data);
    return response;
  }
  
  static async Login(data: object) {
    const response = await this.post(`${this.base_path}/login`, data);
    return response;
  }

  static async Atualizar(id: number, data: object, token: string) {
    const response = await this.put(`${this.base_path}/:id`, data, { params: { id }, headers: {
      'Authorization': `Bearer ${token}`
    } });
    return response;
  }

  static async Deletar(id: number, token: string) {
    const response = await this.delete(`${this.base_path}/:id`, { params: { id }, headers: {
      'Authorization': `Bearer ${token}`
    } });
    return response;
  }

  static async info(token: string) {
    const response = await this.get(`${this.base_path}/info`, {
      headers: {
        'Authorization': `Bearer ${token}`
      }
    });
    return response;
  }

}